Library/Homebrew/os/mac/hardware.rb

```diff
@@ -93,12 +97,15 @@ def arch_64_bit
   # Returns an array that's been extended with ArchitectureListExtension,
   # which provides helpers like #as_arch_flags and #as_cmake_arch_flags.
   def universal_archs
-    # Building 64-bit is a no-go on Tiger, and pretty hit or miss on Leopard.
-    # Don't even try unless Tigerbrew's experimental 64-bit Leopard support is enabled.
-    if MacOS.version <= :leopard && !MacOS.prefer_64_bit?
-      [arch_32_bit].extend ArchitectureListExtension
+    # Support for running on ppc was dropped after Leopard
+    if MacOS.version > :leopard
+      [:x86_64, :i386].extend ArchitectureListExtension
+    # Building 64-bit is pretty hit or miss on Tiger and Leopard.
+    # Don't even try unless Tigerbrew's experimental 64-bit support is enabled.
+    elsif ENV["HOMEBREW_PREFER_64_BIT"]
```

Comment:
At first I did that, but then I found out that the result of `MacOS.prefer_64_bit?` depends on whether the CPU is 64 bits or not. Instead, I wanted to change `universal_archs` so that it only depends on the target operating system (+ environment flags), because a 32-bit system is perfectly capable of building a 64-bit library.

On a side note, it would be interesting to extend `MacOS.prefer_64_bit?` to also allow for building 64-bit binaries on Tiger; I would expect it to be quite unreliable, just like on Leopard, essentially because several of the system libraries only come as i386/ppc, and very few of them are i386/ppc.